python/cudf/cudf/core/udf/masked_lowering.py

```diff
@@ -277,6 +278,11 @@ def masked_scalar_is_null_impl(context, builder, sig, args):
 # else packs it up into a new one that is valid from the get go
 @cuda_lower(api.pack_return, MaskedType)
 def pack_return_masked_impl(context, builder, sig, args):
+    if sig.args[0].value_type is managed_udf_string:
+        struct = cgutils.create_struct_proxy(MaskedType(managed_udf_string))(
+            context, builder, value=args[0]
+        )
+        context.nrt.incref(builder, managed_udf_string, struct.value)
```

Comment:
Why do we incref for a return type that's already masked, but (apparently not) for one that isn't already masked? It feels like there's an asymmetry here that suggests there will be too large or too small a refcount in one of these scenarios - is this handled differently somehow elsewhere?

Reply:
Added the incref for the scalar case in d1ed6458b3f0905fd3941087f5e93465054b503d